x/intertx/types/v1/msg_submit_tx.go

```diff
@@ -63,3 +69,25 @@ func NewMsgSubmitTx(owner string, connectionID string, msg sdk.Msg) *MsgSubmitTx
 		Msg:          anyMsg,
 	}
 }
+
+// PackTxMsgAny marshals the sdk.Msg payload to a protobuf Any type
+func PackTxMsgAny(sdkMsg sdk.Msg) (*codectypes.Any, error) {
+	msg, ok := sdkMsg.(proto.Message)
+	if !ok {
+		return nil, fmt.Errorf("can't proto marshal %T", sdkMsg)
+	}
+
+	any, err := codectypes.NewAnyWithValue(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return any, nil
+}
+
+// UnpackInterfaces implements codectypes.UnpackInterfacesMessage
+func (msg MsgSubmitTx) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
```

Comment:
Without this method, `submit-tx` returning error `cannot unpack Any into type ....`.

Reply:
I think this was an issue with the json example in testing. I was able to successfully test with the fix I submitted in your testing repo. Do we still need this?